cmake/checks/check_02_compiler_features.cmake

```diff
@@ -267,6 +267,7 @@ if( (NOT CMAKE_SYSTEM_NAME MATCHES "CYGWIN") AND
     (NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel") )
   add_flags(CMAKE_REQUIRED_FLAGS "${DEAL_II_CXX_FLAGS_DEBUG}")
   enable_if_supported(DEAL_II_CXX_FLAGS_DEBUG "-Wa,--compress-debug-sections")
+  enable_if_links(DEAL_II_LINKER_FLAGS_DEBUG "-Wl,--compress-debug-sections=zlib")
```

Comment:
Can't you just do
```suggestion
  enable_if_supported(DEAL_II_CXX_FLAGS_DEBUG "-gz")
  enable_if_links(DEAL_II_LINKER_FLAGS_DEBUG "-gz")
```

Reply:
I don't know. What would be the advantage?

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html#Debugging-Options says:
> -gz[=type]
>
>    Produce compressed debug sections in DWARF format, if that is supported. If type is not given, the default type depends on the capabilities of the assembler and linker used. type may be one of ‘none’ (don’t compress debug sections), or ‘zlib’ (use zlib compression in ELF gABI format). If the linker doesn’t support writing compressed debug sections, the option is rejected. Otherwise, if the assembler does not support them, -gz is silently ignored when producing object files.

So my hope was that using this flag gives the compiler a little bit more freedom choosing a compression algorithm but that might be wishful thinking. https://www.phoronix.com/news/GCC-13-gz-Zstd-Debug-Sections also mentions that gcc-13 will allow using `zstd`.